AudioDeviceExample/ViewController.swift

```diff
@@ -202,6 +189,28 @@ class ViewController: UIViewController {
         self.dismissKeyboard()
     }
 
+    // MARK:- IBActions
+    @IBAction func connect(sender: AnyObject) {
+        // Configure access token either from server or manually.
+        // If the default wasn't changed, try fetching from server.
+        if (accessToken == "TWILIO_ACCESS_TOKEN") {
+            TokenUtils.fetchToken(from: tokenUrl) { [weak self]
+                (token, error) in
+                DispatchQueue.main.async {
+                    if let error = error {
+                        let message = "Failed to fetch access token:" + error.localizedDescription
+                        self?.logMessage(messageText: message)
+                        return
+                    }
+                self?.accessToken = token;
+                self?.connectToARoom()
+                }
+            }
+        } else {
+            self.connectToARoom()
+        }
+    }
```

Comment:
I check all the example apps. The disconnect button appears only after token fetching is done. So for the token fetching duration I am disabling the connect button so that it is not called multiple times. Once it transitions to a new view, the user should be able to disconnect even when we are trying to connect to room.

Reply:
Thanks for looking at it! This seems like a reasonable compromise that doesn't add too much complexity.